detox/src/android/espressoapi/DetoxAssertion.js

```diff
@@ -0,0 +1,133 @@
+/**
+
+	This code is generated.
+	For more information see generation/README.md.
+*/
+
+
+const log = require('../../utils/logger').child({ __filename });
+function sanitize_matcher(matcher) {
+  if (!matcher._call) {
+    return matcher;
+  }
+
+  const originalMatcher = typeof matcher._call === 'function' ? matcher._call() : matcher._call;
+  return originalMatcher.type ? originalMatcher.value : originalMatcher;
+} 
+class DetoxAssertion {
+  static assertMatcher(i, m) {
+    if (typeof m !== 'object' || typeof m.constructor !== 'function' || m.constructor.name.indexOf('Matcher') === -1) {
+      const isObject = typeof m === 'object';
+      const additionalErrorInfo = isObject ? typeof m.constructor === 'object' ? 'the constructor is no object' : 'it has a wrong class name: "' + m.constructor.name + '"' : 'it is no object';
+      log.error('m should be an instance of Matcher, got "' + m + '", it appears that ' + additionalErrorInfo);
```

Comment:
Please, try to sort it out - why does it happen in unit tests for `detox` project? Maybe there are outdated inputs/mocks that need updating?

![screenshot from 2018-08-13 09-48-45](https://user-images.githubusercontent.com/1962469/44016576-2b54b87e-9ede-11e8-806a-9ecd172bb01d.png)

Reply:
@noomorph The problem is the way that we assert inputs vs. the way that we use it in the tests. I couldn't find the exact source, but in this test suite we are using either a plain object or a constructor with a different name, but the  assertion tests on the name of the constructor. 

I wanted to have a bit of guidance for our users, that they run into real errors or at least warnings when they are misusing an API, but it might be just the wrong thing or a wrong way. I'm happy to discuss solutions here.

Reply:
>  but in this test suite we are using either a plain object or a constructor with a different name, but the assertion tests on the name of the constructor.

So we just need to fix that in tests, right?

Reply:
Yes. I would be glad if you could give it a try. Just replace the log.error with a throw and get the tests green 🙈